awx/main/models/inventory.py

```diff
@@ -803,6 +803,21 @@ def _get_related_jobs(self):
         return UnifiedJob.objects.non_polymorphic().filter(Q(job__inventory=self.inventory) | Q(inventoryupdate__inventory_source__groups=self))
 
 
+class HostMetrics(models.Model):
+    hostname = models.CharField(primary_key=True, max_length=512)
+    first_automation = models.DateTimeField(auto_now_add=True, null=False)
+    last_automation = models.DateTimeField()
+
+    def host_metrics_update(self, hostname):
```

Comment:
I have created some helper functions here. 
My original logic was to put ```bulk_create()``` and ```bulk_update``` code these helper function. 
But later i realized that i can write it down in the ```awx/main/models/events.py``` file itself. So i will remove it afterwards.

Let me know what you think is a better approach. Writing the ```bulk_create``` and ```bulk_update``` query in ```    def _update_host_summary_from_stats(self, hostnames) ``` or writing those query in ```class HostMetrics```
I will write it accordingly.

Reply:
I like the queries in events.py The reason being that the events.py code is in our callback receiver code path, which needs to be performant. I'd rather have any database queries very clearly in the code path rather than "hidden" away in another class.